### PR TITLE
web: test typescript v5

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

Upgrade Typescript to v5: https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/

## Test plan

CI

## App preview:

- [Web](https://sg-web-vb-typescript-next.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/shared/src/search/query/patternMatcher.test.ts [hunk 1]
```diff
@@ -110,8 +110,6 @@ describe('matchValue', () => {
 
             expect(TestEnum.A as TestEnum).toBeMatchedBy(TestEnum.A)
             expect(TestEnum.A as TestEnum).not.toBeMatchedBy(TestEnum.B)
-            // Shouldn't this throw a type error?
```
Comment (vovakulikov) [inline client/shared/src/search/query/patternMatcher.test.ts:113]: I think I don't fully understand this test case (to be honest don't understand it at all). Is it some sort of invariant testing before real tests?

File: enterprise/dev/ci/internal/ci/operations.go [hunk 2]
```diff
@@ -200,6 +200,10 @@ func addClientLintersForAllFiles(pipeline *bk.Pipeline) {
 		withPnpmCache(),
 		bk.Cmd("dev/ci/pnpm-run.sh lint:js:all"))
 
+	pipeline.AddStep(":eslint: ESLint (web)",
```
Comment (vovakulikov) [inline enterprise/dev/ci/internal/ci/operations.go:203]: Cool, curious does this mean that we still will have a problem with long and possible OOMed eslint job on CI if we change something within the web directory?

**Review by vovakulikov — APPROVED** [review]
Tested locally,  seems like safe update, thanks!

File: enterprise/dev/ci/internal/ci/operations.go (hunk 2 above)
Comment (valerybugakov) [reply enterprise/dev/ci/internal/ci/operations.go:203]: AFAIU, we OOM if we run all the ESLint checks on one machine. With this change, I moved the web package check to a separate worker, and it fixed the issue.

File: dev/foreach-ts-project.sh [hunk 3]
```diff
@@ -29,7 +29,6 @@ DIRS=(
   client/template-parser
   client/testing
   client/vscode
-  client/web
   client/wildcard
   dev/release
 )
```
Comment (philipp-spiess) [inline dev/foreach-ts-project.sh:34]: I wonder if we should rename this script now? It's only use for eslint now but maybe someone wants to use it in the future for something different? 😅

**Review by philipp-spiess — APPROVED** [review]

File: dev/foreach-ts-project.sh (hunk 3 above)
Comment (valerybugakov) [reply dev/foreach-ts-project.sh:34]: Good call! Updated

File: client/shared/src/search/query/patternMatcher.test.ts (hunk 1 above)
Comment (fkling) [reply client/shared/src/search/query/patternMatcher.test.ts:113]: Why did you have to remove this test? I assume it verified that a number doesn't match an enum value. It seems it was supposed to throw a type error but that didn't work as expected...

**Pull request merged**